redfish-core/lib/environment_metrics.hpp

```diff
@@ -68,12 +405,68 @@ inline void handleEnvironmentMetricsGet(
         asyncResp->res.jsonValue["Id"] = "EnvironmentMetrics";
         asyncResp->res.jsonValue["@odata.id"] = boost::urls::format(
             "/redfish/v1/Chassis/{}/EnvironmentMetrics", chassisId);
+
+        getFanSpeedsPercent(asyncResp, *validChassisPath, chassisId);
+        getPowerWatts(asyncResp, chassisId);
+        getPowerLimitWatts(asyncResp);
     };
 
     redfish::chassis_utils::getValidChassisPath(asyncResp, chassisId,
                                                 std::move(respHandler));
 }
 
+inline void handleEnvironmentMetricsPatch(
+    App& app, const crow::Request& req,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisId)
+{
+    if (!redfish::setUpRedfishRoute(app, req, asyncResp))
+    {
+        return;
+    }
+
+    std::optional<nlohmann::json> powerLimitWatts;
+    if (!json_util::readJsonPatch(req, asyncResp->res, "PowerLimitWatts",
```

Comment:
There is a cleaner way to do this now 
`"PowerLimitWatts/ControlMode", controlMode`
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/69924/42/redfish-core/lib/account_service.hpp

Oh well. We can do that when it goes upstream

Reply:
Agreed & changed.